_posts/2018-11-15-using-vault-with-strimzi.md

```diff
@@ -0,0 +1,198 @@
+---
+layout: post
+title:  "Using Vault PKI secrets engine as a CA with the Strimzi Kafka Operator"
+date: 2018-11-15
+author: chris_vest
+---
+
+<!--more-->
+
+# A brief introduction to Vault
+
+If you're here you probably know what [Vault](https://www.vaultproject.io/) is and why it's useful; but for the uninitiated here is a brief description from the creators themselves. The Hashicorp Vault project "secures, stores, and tightly controls access to tokens, passwords, certificates, API keys, and other secrets in modern computing [...]".
+
+So how does Vault fit in with Strimzi and Kafka? Well, Vault has the concept of [`Secrets Engines`](https://www.vaultproject.io/docs/secrets/):
+
+> Secrets engines are components which store, generate, or encrypt data. Secrets engines are incredibly flexible, so it is easiest to think about them in terms of their function. Secrets engines are provided some set of data, they take some action on that data, and they return a result.
+> Some secrets engines simply store and read data - like encrypted Redis/Memcached. Other secrets engines connect to other services and generate dynamic credentials on demand. Other secrets engines provide encryption as a service, totp [time-based one-time password] generation, certificates, and much more.
+
+It is this last point, certificates, which we are going to be looking at. Specifically, we will be using the [Vault PKI Secrets Engine](https://www.vaultproject.io/docs/secrets/pki/index.html) to create a CA, which we will then install instead of using CA certificates and private keys generated by the [Strimzi Cluster Operator](http://strimzi.io/docs/master/#cluster-operator-str).
+
+# SSL, Strimzi & Security
```

Comment:
Maybe the main title could be "Securing a Kafka cluster on Kubernetes" ? In this way the first paragraph talks about the "incubator" way which seems not to be good the followed by the Strimzi support.